### PR TITLE
Enable range positions (`-Yrangepos`) by default

### LINKED ISSUE #472: Enable range positions (-Yrangepos) by default
this isn't a new idea, but it hasn't had its own ticket before, at least, I can't find one.

at the umbrella ticket https://github.com/scala/scala-dev/issues/430#issuecomment-332562635 @adriaanm said (in response to @xeno-by's question) rangepos "will be on by default" in 2.13, as a goal at least

@olafurpg has said "The Scalameta Semantic API, which Scalafix uses, requires -Yrangepos" ([here](https://github.com/scala/scala-dev/issues/390#issuecomment-303989029)) so this is an important area for anyone interested in improving the quality of tooling for Scala

some relevant links:

* https://github.com/scala/bug/labels/positions
* https://github.com/scala/scala-dev/issues/390

### PR DESCRIPTION
**Pull request by eed3si9n**
This mainly affects users of Scalameta SemanticDB, which previously required adding `-Yrangepos` to `scalacOptions`. (But continuing to supply the flag, for example when cross-building to 2.12, is harmless.)

Note that for a long time now, Metals has enabled `-Yrangepos` on users' behalf. So Metals users won't notice any difference, and the fact that so many users have had `-Yrangepos` enabled for them behind the scenes increases our confidence that the flag is robust.

Fixes https://github.com/scala/scala-dev/issues/472

### CODE REVIEW

**Comment by hrhino** [comment]
Wait, this is `reflect.runtime.Settings`... I think you (also) need to change the version in `scala.tools.nsc.settings.Settings` for the compiler flag.

**Comment by eed3si9n** [comment]
Yea. That makes sense!

**Comment by eed3si9n** [comment]
```
!!    1 - run/StubErrorTypeDef.scala                [output differs]
!!    2 - run/macro-rangepos-args                   [output differs]
!!    3 - run/sd187.scala                           [output differs]
!!    4 - run/string-switch-pos.scala               [output differs]
!!    5 - run/t6288.scala                           [output differs]
!!    6 - run/t7271.scala                           [output differs]
!!    8 - run/t7331a.scala                          [output differs]
!!    7 - run/t7331c.scala                          [output differs]
!!   10 - run/typetags_without_scala_reflect_typetag_lookup.scala  [output differs]
!!    9 - run/typetags_without_scala_reflect_typetag_manifest_interop.scala  [output differs]
```

**Comment by SethTisue** [comment]
@eed3si9n is this ready for a trip through the community build?

**Comment by eed3si9n** [comment]
I think so.

**Comment by SethTisue** [comment]
~~queued: https://scala-ci.typesafe.com/job/scala-2.13.x-integrate-community-build/3624/~~
~~queued: https://scala-ci.typesafe.com/job/scala-2.13.x-integrate-community-build/3625/~~
queued: https://scala-ci.typesafe.com/job/scala-2.13.x-integrate-community-build/3638/

**Comment by SethTisue** [comment]
run 3638 was green

**Comment by eed3si9n** [comment]
Nice!

File: test/files/run/macro-rangepos-args.check [hunk 1]
```diff
@@ -1 +1 @@
-Line: 3. Width: 1.
+Line: 3. Width: 5.
```
Comment (dwijnand) [inline test/files/run/macro-rangepos-args.check:1]: Reverts https://github.com/scala/scala/commit/0412783b5cfa64781708e55757ff9fbfe6408266#diff-0298f348965a4dcbd2f8e4b61ae8ab22, so all good.

File: test/files/run/string-switch-pos.check [hunk 2]
```diff
@@ -1,63 +1,63 @@
 [[syntax trees at end of                    patmat]] // newSource1.scala
-[6]package [6]<empty> {
-  [6]class Switch extends [13][187]scala.AnyRef {
+[0:187]package [0:0]<empty> {
```
Comment (dwijnand) [inline test/files/run/string-switch-pos.check:2]: That `[0:0]` looks wrong... but doesn't seem to affect anything.

**Review by dwijnand — APPROVED** [review]
LGTM.  Any trailing reason this is draft?

**Comment by eed3si9n** [comment]
> Any trailing reason this is draft?

@SethTisue set to draft initially presumably because the build was still failing and it was unproven at the time. Now that it's passing on Community Build, I think it's ready for review / discussion.

**Comment by SethTisue** [comment]
as already discussed with @lrytz, I think we should bite the bullet on this for 2.13.4

**Comment by SethTisue** [comment]
~~as already approved (modulo details, I mean) by @lrytz, I think we should bite the bullet on this for 2.13.4~~

~~everyone using Metals is running with range positions on, and Metals has been pretty popular for a while now, so that gives substantial additional confidence this is safe enough to run with~~

oh haha I already said that stuff at https://github.com/scala/scala-dev/issues/472. but I still believe it :-)

also there I said "an upcoming 2.13.x release" but now we're saying 2.13.4 specifically

**Comment by SethTisue** [comment]
publicized: https://contributors.scala-lang.org/t/range-positions-yrangepos-will-be-on-by-default-in-2-13-4/4506

**Pull request merged**